src/middleware/errorHandler.ts

```
@@ -0,0 +1,514 @@
+  next: NextFunction
+): void {
+  // Handle PostgreSQL specific errors
+  if (error.code) {
+    switch (error.code) {
+      case '23505': // Unique constraint violation
+        const constraintError = new BaseError(
+          'Resource already exists',
+          409,
+          'RESOURCE_EXISTS',
+          { constraint: error.constraint, detail: error.detail }
+        );
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)